### PR TITLE
Proxy Support

### PR DESCRIPTION
**Pull request by reedrosenbluth**
Usage:

```javascript
var remote = new Remote({
  proxy: 'http://0.0.0.0:3128',
  servers: [ 'wss://s1.ripple.com:443' ]
});
```

### CODE REVIEW

File: src/core/server.js [hunk 1]
```diff
@@ -436,7 +437,18 @@ Server.prototype.connect = function() {
     log.info(this.getServerID(), 'connect');
   }
 
-  const ws = this._ws = new WebSocket(this._opts.url);
+  if (this._remote.hasOwnProperty('proxy')) {
+    let parsed = url.parse(this._opts.url);
+    let opts = url.parse(this._remote.proxy);
+    opts.secureEndpoint = parsed.protocol ? parsed.protocol === 'wss:' : false;
```
Comment (clark800) [inline src/core/server.js]: This simplifies to `parsed.protocol === 'wss:'`

Comment (reedrosenbluth) [inline src/core/server.js]: fixed

**Comment by geertweening** [comment]
What about username and password support?

**Comment by reedrosenbluth** [comment]
@geertweening 

```javascript
var remote = new Remote({
  proxy: 'http://username:password@0.0.0.0:3128',
  servers: [ 'wss://s1.ripple.com:443' ]
});
```

**Comment by geertweening** [comment]
:+1:

**Comment by clark800** [comment]
LGTM

**Pull request merged**